keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md

```diff
@@ -166,46 +235,54 @@ grpc stream for each of them. This way requesting progress notification on a
-The bug was only fixed in v3.4.25 and v3.5.8 (2 months old), and could cause
-client missing an event.
-
-For Alpha feature will be only available under a feature gate, and we will
-depend on documenting the minimal required etcd version in feature gate
-description.
-
 [etcd-io/etcd#15220]: https://github.com/etcd-io/etcd/issues/15220
 
-<<[UNRESOLVED @serathius]>>
-For Beta propose how kube-apiserver should behave if user is running 
-older/affected etcd version.
-
-Options for Beta:
-* Ask user to proviede etcd version to kube-apiserver `--storage-backend=etcd3.4.25`
-* Make the feature opt-in with flag `--allow-using-progress-notify`
-* Have kube-apiserver check cluster version in etcd `/version` endpoint.
-  Retry the check logic if `WatchProgressRequest` fails.
-* Fallback to reading from etcd if no progress notification within `X` seconds.
-<<[/UNRESOLVED]>>
-
 ### What if the watch cache is stale?
 
-This design requires wait for a watch cache to catch up to the needed revision
-for consistent reads. If the cache doesn't catch up within some time limit we
-either fail the request or have a fallback.
+This design requires wait for a watch cache to catch up to the needed revision.
+There might be situation where watch is not providing updates causing watch cache to be permanently stale.
+For example if watch stream is clogged, or when using etcd version v3.3 and older that don't provide progress notifications.
+
+If the watch cache doesn't catch up in within some time limit we either fail the request or have a fallback.
 
 If the fallback is to forward consistent reads to etcd, a cascading failure
 is likely to occur if caches become stale and a large number of read requests
 are forwarded to etcd.
 
 Since falling back to etcd won't work, we should fail the requests and rely on
 rate limiting to prevent cascading failure.  I.e. `Retry-After` HTTP header (for
-well behaved clients) and [Priority and Fairness](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md).
+well-behaved clients) and [Priority and Fairness](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md).
+
+For such situations we will provide users with following tools:
+* a dedicated `apiserver_watch_cache_read_wait` metric to detect a problem with
+  watch cache.
+* a per-request override to disable watch cache to allow debugging.
+
+Metric `apiserver_watch_cache_read_wait` will measure wait time experienced by 
+reads for watch cache to become fresh. If user notices a latency request in
+they can use this metric to confirm that the issue is caused by watch cache.
+
+Per request override should allow user to compare request results without
+impacting other requests or requiring to redeploy whole cluster. The exact
+details of override API will be clarified during API review. In healthy
```

Comment:
IIUC @deads2k wants the ListOptions field, but I would leave it to him to approve during API review.